### DISCUSSION TITLE
Sorting data within stacked bar chart groups

### DISCUSSION THREAD

**Discussion by amory1989** [iTwin/itwinjs-core#3566] [Q&A] [answered]
Hello! Huge fan of ChartJS - huge thanks to the people who made this. 

I have a stacked bar chart with groups. In the attached example, there are two "groups" of stacked bars (they are in fact stacked floating bars - hacked it to make it look like box and whiskers). I want to be able to sort the data *within each group* from lowest to highest bar mid point. 

I can do this for one of the groups by just sorting my whole chart.data.datasets array by the midpoint of each of the first group's bars... but this sorts *all my data* (including for the other group), when I really want to be able to sort the data for each group separately. 

A single row of of my chart.data.datasets array looks like this: 

[[lowerBound group1, upperBound group1],[lowerBoundGroup2, upperBoundGroup2]] 

And Chart.js by default sorts all the data according to the order of this parent array, chart.data.datasets. Is there some built-in way to sort within groups? And, if there's not a built in way, any ideas about how I could pull it off? 

Thanks in advance for whatever guidance you all might be able to provide - and thanks again for a fantastic data visualization resource. 

<img width="805" alt="Screen Shot 2022-04-28 at 2 59 51 PM" src="https://user-images.githubusercontent.com/104592691/165826739-779cbb67-ea9c-4cb9-82fb-2b566b3176dc.png">

**Comment by MichaelBelousov** [comment, marked as answer]
This seems like you're expecting to be talking to developers of Chart.js. They have a [github discussions page](https://github.com/chartjs/Chart.js/discussions) here. Did you intend to ask a question related to iTwin.js or chart.js? I don't know who here has a lot of experience with chart.js.

**Reply by amory1989** [reply]
Oops! You are entirely correct. My bad. Thank you :)
